File: src/main/java/me/pafias/gambling/Gambling.java
package me.pafias.gambling;

import net.milkbowl.vault.economy.Economy;
import org.bukkit.entity.Player;
import org.bukkit.plugin.RegisteredServiceProvider;
import org.bukkit.plugin.java.JavaPlugin;

public final class Gambling extends JavaPlugin {
    private static Gambling plugin;
    public static Gambling get() {
        return plugin;
    }

    private static Economy economy = null;
    public static Economy getEconomy() {
        return economy;
    }

    @Override
    public void onEnable() {
        plugin = this;

        if (!setupEconomy())
            getLogger().severe("No Vault or economy provider found!");

        getCommand("blackjack").setExecutor(new BlackjackCommand());
    }

    @Override
    public void onDisable() {
        getServer().getOnlinePlayers().forEach(Player::closeDialog);
    }

    private boolean setupEconomy() {
        if (!getServer().getPluginManager().isPluginEnabled("Vault"))
            return false;

        RegisteredServiceProvider<Economy> rsp = getServer().getServicesManager().getRegistration(Economy.class);
        if (rsp == null)
            return false;

        economy = rsp.getProvider();
        return economy != null;
    }

}

File: src/main/kotlin/me/pafias/gambling/Gambling.kt
package me.pafias.gambling

import net.milkbowl.vault.economy.Economy
import org.bukkit.entity.Player
import org.bukkit.plugin.java.JavaPlugin

class Gambling : JavaPlugin() {

    lateinit var plugin: Gambling
    lateinit var economy: Economy

    override fun onEnable() {
        plugin = this

        if (!setupEconomy()) logger.severe("No Vault or economy provider found!")

        getCommand("blackjack")!!.setExecutor(BlackjackCommand())
    }

    override fun onDisable() {
        server.onlinePlayers.forEach { obj: Player? -> obj!!.closeDialog() }
    }

    private fun setupEconomy(): Boolean {
        if (!server.pluginManager.isPluginEnabled("Vault")) return false

        val rsp = server.servicesManager.getRegistration<Economy?>(Economy::class.java) ?: return false

        economy = rsp.provider
        return true
    }

    fun get(): Gambling {
        return plugin
    }
}
